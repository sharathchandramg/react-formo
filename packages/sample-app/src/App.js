import React, { Component } from 'react';
import './App.css';
import Form from 'react-web-formo';

const data = require('./schema/22-checklist.json');

const testData = [
  {
    type: 'phone',
    name: 'phone_number',
    label: 'Phone Number',
    required: true,
    editable: true,
    hidden: false,
  },
];

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      fields: data,
      formData: {},
    };
  }

  render() {
    return (
      <div className="App">
        <div style={{ width: '85%' }}>
          <Form
            className="form"
            ref={c => (this.formGenerator = c)}
            fields={this.state.fields}
            formData={this.state.formData}
          />
        </div>
        <div className="buttons" style={{ justifyContent: 'center' }}>
          <button
            className="button subBtn"
            type="button"
            onClick={() => this.formGenerator.getValues()}
          >
            Submit
          </button>
          <button
            className="button clrBtn"
            type="reset"
            onClick={() => this.formGenerator.resetForm()}
          >
            Clear Form
          </button>
          <button
            className="button vldBtn"
            type="button"
            onClick={() => this.formGenerator.onValidateFields()}
          >
            Validate Form
          </button>
        </div>
      </div>
    );
  }
}

export default App;
