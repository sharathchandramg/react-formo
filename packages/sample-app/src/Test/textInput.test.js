import React from 'react';
import { shallow, mount } from 'enzyme';
import App from '../App';

describe(App, () => {
    it('Renders App', () => {
        const app = shallow(<App />);
        expect(App).toMatchSnapshot();
    });

    it('should call function when submit button is clicked', () => {
        const wrapper = mount(<App />);
        const submitBtn = wrapper.find('button.subBtn');
        submitBtn.simulate('click');
        const text = wrapper.find('#error').text();
        expect(text).toEqual('Phone Number is required');
    });

    it('should call function when validate button is clicked', () => {
        const wrapper = mount(<App />);
        const validateBtn = wrapper.find('button.vldBtn');
        validateBtn.simulate('click');
        const text = wrapper.find('#error').text();
        expect(text).toEqual('Phone Number is required');
    });

    it('should validate the input when submit button is clicked', () => {
        const newValue = 'testing component';
        const wrapper = mount(<App />);
        const input = wrapper.find('input');
        input.simulate('change', { target: { value: newValue } });
        const submitBtn = wrapper.find('button.subBtn');
        submitBtn.simulate('click');
        const text = wrapper.find('#error').text();
        expect(text).toEqual('Phone Number should be valid mobile number');
    });
});
