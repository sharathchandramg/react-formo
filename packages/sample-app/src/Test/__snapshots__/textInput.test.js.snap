// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders App 1`] = `[Function]`;
